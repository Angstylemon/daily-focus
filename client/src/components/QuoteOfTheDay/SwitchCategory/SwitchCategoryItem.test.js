import { jssPreset } from "@material-ui/styles";
import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { shallow } from "enzyme";

import Item from "./item.js";

let valueState;
const setValue = (newValue) => {
    valueState = newValue;
};

beforeEach(() => {
    valueState = "";
});

test("Selected item renders correctly", () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshotComponent = shallowRenderer.render(<Item text="Text" selected={true} />);
    expect(snapshotComponent).toMatchSnapshot();
});

test("Unselected item renders correctly", () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshotComponent = shallowRenderer.render(<Item text="Text" selected={false} />);
    expect(snapshotComponent).toMatchSnapshot();
});

test("When an item is clicked, the category is changed", () => {
    const categoryValue = "value";
    const wrapper = shallow(
        <Item text="Text" value={categoryValue} selected={false} clickHandler={setValue} />
    );
    wrapper.find(`#quoteCategoryValue-${categoryValue}`).simulate("click");
    expect(valueState).toBe(categoryValue);
});
