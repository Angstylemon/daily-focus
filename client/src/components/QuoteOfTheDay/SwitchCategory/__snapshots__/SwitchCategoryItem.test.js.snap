// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selected item renders correctly 1`] = `
<div
  className="selectedCategoryOption"
  id="quoteCategoryValue-undefined"
  onClick={[Function]}
>
  Text
  <Memo />
</div>
`;

exports[`Unselected item renders correctly 1`] = `
<div
  className="categoryOption"
  id="quoteCategoryValue-undefined"
  onClick={[Function]}
>
  Text
  <Memo />
</div>
`;
